dtm0/it/s3setup/s3setup.sh

```diff
@@ -0,0 +1,327 @@
+
+M0D_FIDS_HEX=()
+M0D_PIDS=()
+S3_FID_HEX=
+S3_PID=
+
+NO_BOOTSTRAP=0
+RUN_S3BENCH=0
+RUN_S3WORKLOAD=0
+
+function repos_clone()
+{
+    ssh-keygen -F github.com || ssh-keyscan github.com >> ~/.ssh/known_hosts
+
+    pushd $DEVEL_DIR
+
+    git clone --recurse-submodules https://github.com/Seagate/cortx-motr.git
+    pushd cortx-motr
+    git checkout dtm0-main
+    popd
+
+    git clone https://github.com/Seagate/cortx-hare.git
+    pushd cortx-hare
+    git checkout dtm-0
+    popd
+
+    popd
+}
+
+function cfgs_backup()
+{
+    cp $CLUSTER_YAML $CLUSTER_YAML_BAK
+    cp $MOTR_SYSCONFIG $MOTR_SYSCONFIG_BAK
+    cp $LNET_CONF $LNET_CONF_BAK
+    mv $PIP_CONF $PIP_CONF_BAK
+}
+
+function cfgs_restore()
+{
+    cp $CLUSTER_YAML_BAK $CLUSTER_YAML
+    cp $MOTR_SYSCONFIG_BAK $MOTR_SYSCONFIG
+    cp $LNET_CONF_BAK $LNET_CONF
+    # Uncomment and change MOTR_DEVEL_WORKDIR_PATH in motr sysconfig
+    # to use development files.
+    sed -i "s|^#MOTR_DEVEL_WORKDIR_PATH=.*|MOTR_DEVEL_WORKDIR_PATH=${MOTR_DIR}|" $MOTR_SYSCONFIG
+    patch $CLUSTER_YAML < $CLUSTER_YAML_PATCH
+}
+
+function motr_build()
+{
+    pushd $MOTR_DIR
+    ./scripts/install-build-deps
+    MAKE_OPTS=$MOTR_MAKE_OPTS CONFIGURE_OPTS=$MOTR_CONFIGURE_OPTS ./scripts/m0 rebuild
+    popd
+}
+
+function hare_build()
+{
+    pushd $HARE_DIR
+    patch -p1 < $HARE_PATCH
+    make
+    popd
+}
+
+function motr_install()
+{
+    pushd $MOTR_DIR
+    ./scripts/install-motr-service -l
+    rm -f /usr/lib64/libmotr-helpers.so.0 /usr/lib64/libmotr.so.2
+    ln -s "${MOTR_DIR}/helpers/.libs/libmotr-helpers.so.0" /usr/lib64/libmotr-helpers.so.0
+    ln -s "${MOTR_DIR}/motr/.libs/libmotr.so.2" /usr/lib64/libmotr.so.2
+    ln -s "${MOTR_DIR}/extra-libs/galois/src/.libs/libgalois-0.1.so" /usr/lib64/libgalois-0.1.so
+    popd
+}
+
+function hare_install()
+{
+    pushd $HARE_DIR
+    make devinstall    
+    popd
+}
+
+function cluster_bootstrap()
+{
+    hctl bootstrap --mkfs $CLUSTER_YAML
+}
+
+function cluster_stop()
+{
+    hctl shutdown
+}
+
+function s3bench_run()
+{
+    local ak
+    local sk
+    local sgiam_pass
+
+    # Generate password.
+    sgiam_pass=$(s3cipher decrypt --data="$(s3confstore properties:///opt/seagate/cortx/auth/resources/authserver.properties getkey --key ldapLoginPW)" --key="$(s3cipher generate_key --const_key cortx)")
+
+    # Fetch the user from ldap, copy access key (ak) and secret key (sk) from output.
+    read ak sk <<< $(ldapsearch -b "ou=accesskeys,dc=s3,dc=seagate,dc=com" -x -w $sgiam_pass -D "cn=sgiamadmin,dc=seagate,dc=com" "(&(objectclass=accesskey))" | egrep "^ak:|^sk:" | awk '{ print $2; }')
+
+    # Get s3bench.
+    if [[ ! -e ./s3bench ]]; then
+        wget $S3BENCH_URL -O ./s3bench
+        chmod +x ./s3bench
+    fi
+
+    # Run s3bench.
+    ./s3bench -accessKey $ak -accessSecret "${sk}" -bucket test1 -endpoint http://127.0.0.1  -numClients 4 -numSamples 4 -objectSize 1Mb
+}
+
+function fids_hex_get()
```

Comment:
this foo() looks so similar to all2all's. can it be generalized?

Reply:
Done.